src/platforms/linux/daemon/apptracker.cpp

```diff
@@ -104,59 +91,116 @@ void AppTracker::userRemoved(uint userid) {
   QDBusConnection::disconnectFromBus("user-" + QString::number(userid));
 }
 
-void AppTracker::gtkLaunchEvent(const QByteArray& appid, const QString& display,
-                                qlonglong pid, const QStringList& uris,
-                                const QVariantMap& extra) {
-  Q_UNUSED(display);
-  Q_UNUSED(uris);
-  Q_UNUSED(extra);
+// static
+// Expand unicode escape sequences.
+QString AppTracker::decodeUnicodeEscape(const QString& str) {
+  static const QRegularExpression re("(\\\\x[0-9A-Fa-f][0-9A-Fa-f])");
+
+  QString result = str;
+  qsizetype offset = 0;
+  while (offset < result.length()) {
+    QRegularExpressionMatch match = re.match(result, offset);
+    if (!match.hasMatch()) {
+      break;
+    }
+
+    bool okay;
+    qsizetype start = match.capturedStart(0);
+    QChar code = match.captured(0).mid(2).toUShort(&okay, 16);
+    if (okay && (code != 0)) {
+      result.replace(start, match.capturedLength(0), QString(code));
+      offset = start + 1;
+    } else {
+      offset = match.capturedEnd(0);
+    }
+  }
+
+  return result;
+}
 
-  QString appIdName(appid);
-  while (appIdName.endsWith('\0')) {
-    appIdName.chop(1);
+// The naming convention for snaps follows one of the following formats:
+//   snap.<pkg>.<app>.service - assigned by systemd for services
+//   snap.<pkg>.<app>-<uuid>.scope - transient scope for apps
+//   snap.<pkg>.hook.<app>-<uuid>.scope - transient scope for hooks
+//
+// However, at some point the separator between the app and UUID was
+// swapped from a dot to a dash. Which makes the parsing a bit of a pain.
+//
+// See: https://github.com/snapcore/snapd/blob/master/sandbox/cgroup/scanning.go
+QString AppTracker::snapDesktopId(const QString& scope) {
```

Comment:
Should these string manipulation functions be moved to LinuxDependencies or some String utils file since they don't need any class state? Also, consider adding some tests for the string manipulation functions, now or later.